crates/web5/src/credentials/verifiable_credential_1_1.rs

```diff
@@ -383,57 +359,127 @@ impl VerifiableCredential {
             credential_subject: vc_credential_subject,
         };
 
-        validate_vc_data_model(&vc)?;
+        if verify {
+            validate_vc_data_model(&vc)?;
+        }
 
         Ok(vc)
     }
+
+    fn sign_with_signer(&self, key_id: &str, signer: Arc<dyn Signer>) -> Result<String> {
```

Comment:
interesting name. not pub though? are we not allowing a generic signer after all?

Reply:
also, why do we need key_id for this one?

Reply:
due to some reasonable doubts, we're moving forward with solely "sign with did" on the public api surface at this time. we can extend later by adding lower level support for DSA signers and JOSE signers, but for now they aren't strictly necessary.

a `Signer` is a DSA signer, which is distinct from a (proposed) `JoseSigner`. DSA is lower level, it's just about producing a signature for a given input, whereas JOSE is higher level which includes additional semantic data, in our case, a `kid`.